### PR TITLE
Support --data-dir commandline argument

### PR DESCRIPTION
**Pull request by sumof2primes**
--data-dir default ./
[AUTOMATIC1111/stable-diffusion-webui/wiki/Command-Line-Arguments-and-Settings](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Command-Line-Arguments-and-Settings)

### CODE REVIEW

File: scripts/tag_autocomplete_helper.py [hunk 1]
```diff
@@ -4,13 +4,14 @@
...
 from modules import scripts, script_callbacks, shared, sd_hijack
+from modules.paths import data_path
 import yaml
 
 # Webui root path
 FILE_DIR = Path().absolute()
 
 # The extension base path
-EXT_PATH = FILE_DIR.joinpath('extensions')
+EXT_PATH = Path(data_path).joinpath('extensions')
```
Comment (DominikDoom) [inline scripts/tag_autocomplete_helper.py:14]:
```suggestion
EXT_PATH = Path(data_path).joinpath('extensions').absolute()
```

File: scripts/tag_autocomplete_helper.py [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
 
 # Tags base path
 TAGS_PATH = Path(scripts.basedir()).joinpath('tags')
```
Comment (DominikDoom) [inline scripts/tag_autocomplete_helper.py:17]:
```suggestion
TAGS_PATH = Path(scripts.basedir()).joinpath('tags').absolute()
```

**Review by DominikDoom — CHANGES_REQUESTED** [review]
This does not work on Windows with just what you changed here, you need to make the main paths either all absolute or all relative. At least in the case of --data-dir being a relative path, which it is by default. Else path comparisons will fail.

It will also only work if your data dir is a subfolder of the webui itself, if it is outside of it or even on a different drive the script won't work at all in the first place since Gradio doesn't load the javascript for security reasons.

**Comment by sumof2primes** [comment]
Ok. I understand --data-dir won't work expected for security reasons.
It's better to use below options separately for userdata.
--ckpt-dir
--vae-dir
--lora-dir
--embeddings-dir

**Pull request closed without merging**